app/jobs/regular/emit_web_hook_event.rb

```diff
@@ -71,71 +52,164 @@ def web_hook_request(args, web_hook)
+    end
 
-      retry_web_hook if response&.status != 200
+    def ping_event?(event_type)
+      PING_EVENT == event_type.to_s
     end
 
-    def retry_web_hook
-      if SiteSetting.retry_web_hook_events?
-        @orig_args[:retry_count] = (@orig_args[:retry_count] || 0) + 1
-        return if @orig_args[:retry_count] > MAX_RETRY_COUNT
-        delay = RETRY_BACKOFF**(@orig_args[:retry_count] - 1)
-        Jobs.enqueue_in(delay.minutes, :emit_web_hook_event, @orig_args)
+    def webhook_inactive?
+      !web_hook.active?
+    end
+
+    def group_webhook_invalid?
+      web_hook.group_ids.present? && (group_id.present? ||
+        !web_hook.group_ids.include?(group_id))
+    end
+
+    def category_webhook_invalid?
+      web_hook.category_ids.present? && (!category_id.present? ||
+        !web_hook.category_ids.include?(category_id))
+    end
+
+    def tag_webhook_invalid?
+      web_hook.tag_ids.present? && (tag_ids.blank? ||
+        (web_hook.tag_ids & tag_ids).blank?)
+    end
+
+    def arguments
+      @arguments
+    end
+
+    def event_type
+      @event_type ||= arguments[:event_type].to_s
+    end
+
+    def event_name
+      @event_name ||= arguments[:event_name].to_s
+    end
+
+    def retry_count
+      @retry_count ||= @arguments[:retry_count]
+    end
+
+    def webhook_id
+      @webhook_id ||= @arguments[:web_hook_id]
+    end
+
+    def group_id
+      @group_id ||= @arguments[:group_id]
+    end
+
+    def category_id
+      @category_id ||= @arguments[:category_id]
+    end
+
+    def tag_ids
+      @tag_ids ||= @arguments[:tag_ids]
+    end
+
+    def payload
+      @payload ||= @arguments[:payload]
+    end
+
+    def parsed_payload
+      @parsed_payload ||= JSON.parse(payload)
+    end
+
+    def web_hook
+      @web_hook ||= WebHook.find_by(id: webhook_id)
+    end
+
+    def build_webhook_headers(uri, web_hook_body, web_hook_event)
+      content_type =
+        case web_hook.content_type
+        when WebHook.content_types['application/x-www-form-urlencoded']
+          'application/x-www-form-urlencoded'
+        else
+          'application/json'
+        end
+
+      headers = {
+        'Accept' => '*/*',
+        'Connection' => 'close',
+        'Content-Length' => web_hook_body.bytesize,
+        'Content-Type' => content_type,
+        'Host' => uri.host,
+        'User-Agent' => "Discourse/#{Discourse::VERSION::STRING}",
+        'X-Discourse-Instance' => Discourse.base_url,
+        'X-Discourse-Event-Id' => web_hook_event.id,
+        'X-Discourse-Event-Type' => event_type
+      }
+
+      headers['X-Discourse-Event'] = event_name if event_name.present?
+
+      if web_hook.secret.present?
+        headers['X-Discourse-Event-Signature'] = "sha256=#{OpenSSL::HMAC.hexdigest("sha256", web_hook.secret, web_hook_body)}"
       end
+
+      headers
     end
+
+    def build_webhook_body
+      body = {}
+
+      if ping_event?(event_type)
+        body[:ping] = 'OK'
```

Comment:
Are we mixing string and symbol keys in the same body? We should pick one type and stick with it.